File: solutions/Arrays/0026-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& arr) {
        int n=arr.size();
        int i=0,j=0;
        while(j<n){
            int val=arr[j];
            arr[i++]=val;
            while(j<n && arr[j]==val){
                j++;
            }
        }
        return i;
    }
};

File: solutions/Arrays/0026-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int slow=0,fast=0;
        while(fast<nums.length){
            if(fast==0 || nums[fast]!=nums[fast-1]){
                nums[slow++]=nums[fast++];
            }else{
                fast++;
            }
        }
        return slow;
    }
}

/*

1 1 1 1 2 2 2

*/